File: API/Domain/FoodDiaries/FoodDiary.cs
using Domain.Foods;

namespace Domain.FoodDiaries;

public record FoodDiary(
    Guid Id,
    Guid UserId,
    DateOnly Date,
    Meal? Breakfast,
    Meal? Lunch,
    Meal? Snack,
    Meal? Dinner,
    FoodNutriments TotalNutriments,
    FoodEnergy TotalEnergy
);

public record Meal(
    ICollection<EatenFood> EatenFoods,
    FoodNutriments TotalNutriments,
    FoodEnergy TotalEnergy
);

public record EatenFood(
    Food Food,
    int Grams
);

File: client/src/app/tabs/features/food-diary-tab/types/FoodDiary.ts
import {FoodNutriments} from "./FoodNutriments";
import {FoodEnergy} from "./FoodEnergy";
import {FoodMeal} from "./FoodMeal";

export type FoodDiary = {
  id: string;
  userId: string;
  date: string;
  breakfast: FoodMeal[];
  lunch: FoodMeal[];
  snack: FoodMeal[];
  dinner: FoodMeal[];
  totalNutriments: FoodNutriments;
  totalEnergy: FoodEnergy;
}
